pyfar/dsp/dsp.py

```diff
@@ -1487,4 +1494,314 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1):
+
+    ``signal_normalized = signal * target / reference``,
+
+    where `target` equals
+    1 and `reference` is the maximum absolute amplitude before the
+    normalization.
+
+    Several normalizations are possible, which in fact are different ways
+    of computing the `reference` value (e.g. based on the spectrum).
+    See the parameters for details.
+
+    Parameters
+    ----------
+    signal: Signal, TimeData, FrequencyData
+        Input signal of the signal class.
+    domain: string
+        Domain of the data to compute the reference value.
+
+        ``'time'``
+           Normalization based on the time data.
+        ``'freq'``
+           Normalization based on the frequency data. In this case the
+           magnitudespectrum with FFT normalization is used
+           (``signal.freq``, cf. :py:mod:`FFT concepts <pyfar._concepts.fft>`).
+
+        The default is ``'time'``.
+    dB: bool, optional
+        If `dB` is ``True``, the data is logarithmized according to
+        :py:func:`~pyfar.dsp.decibel` and the signal's properties before
+        the calculation of the reference value.
+        The default is ``False``.
+    operation: string, optional
+        Operation per channel to compute the `reference` value described above
+        (according to ``domain`` and ``dB``).
+
+        ``'max'``
+            Takes the maximum absolute value.
+        ``'mean'``
+            Takes the mean of the absolute values.
+        ``'energy'``
+            Takes the mean of the absolute values.
+        ``'power'``
+            Takes the mean of the absolute values.
+        ``'rms'``
+            Takes the root mean square value.
+
+        The default is ``'max'``.
+    channel_handling: string, optional
+        Define how multi-channel signals are handled. This parameter does
+        not affect single-channel signals.
+
+        ``'each'``
+            Separate normalization of each channel.
+        ``'max'``
+            Normalize to the maximum `reference` value across channels.
+        ``'min'``
+            Normalize to the minimum `reference` value across channels.
+        ``'mean'``
+            Normalize to the mean `reference` value across the channels.
+
+       The default is ``'max'``
+    target: scalar, array
+        The target to which the signal is normalized in the unit according to
+        `dB`. Can be a scalar or an array. In the latter case the shape of
+        `target` must be broadcasted ``signal.cshape``. The default is 0 dB if
+        ``dB=True`` and 1 otherwise.
+    freq_range: tuple, array_like, double
+        Two element vector specifying upper and lower frequency in Hz between
+        which the `reference` value for normalization is computed. This will
+        just effect if the normalization is based on the frequency data.
+        Therefore, `freq_range` won't be used if ``domain='time'``.
+        The default is ``None``, which will use the whole frequency range of
+        the signal.
+
```

Comment:
Docs for ``unit`` and ``return_reference`` to be added here.
The docstring for unit could contain a short notice that the ``find_nearest...`` functions are used. I'm not sure where to put the explanation, that for ``unit=samples`` the second element in ``limits`` defines the first sample excluded. To limits, units or in the Notes?